### PR TITLE
feat(widgets): construction sites as news

### PR DESCRIPTION
**Pull request by digorath**
## Changes proposed in this PR:

- added new widget for construction sites which are represented as newsitems

---

SVA-274

### CODE REVIEW

File: src/screens/DetailScreen.js [hunk 1]
```diff
@@ -67,6 +69,11 @@ export const DetailScreen = ({ navigation, route }) => {
     isConnected && auth();
   }, []);
 
+  useEffect(() => {
+    const options = screenOptionsWithShare(true)({ navigation, route });
```
Comment (donni106) [inline src/screens/DetailScreen.js:73]: crazy

File: src/screens/DetailScreen.js [hunk 2]
```diff
@@ -67,6 +69,11 @@ export const DetailScreen = ({ navigation, route }) => {
...
+    !bookmarkable && navigation.setOptions(options);
```
Comment (donni106) [inline src/screens/DetailScreen.js:74]: can you please make a more detailed sentence?
if not bookmarkable, show the hamburger? headerRight true brings me there, right?

Comment (digorath) [reply src/screens/DetailScreen.js:78]: see above, the true was incorrect. i just forgot to put in the correct value.

File: src/screens/IndexScreen.js [hunk 3]
```diff
@@ -34,6 +34,7 @@ export const IndexScreen = ({ navigation, route }) => {
   const query = route.params?.query ?? '';
   const title = route.params?.title ?? '';
   const titleDetail = route.params?.titleDetail ?? '';
+  const bookmarkable = route.params?.bookmarkable;
```
[the remark this answers is not in the document]
Comment (digorath) [reply src/screens/IndexScreen.js:37]: yes. the fallback ist being added at the detail screen level

File: src/screens/DetailScreen.js (hunk 2 above)
Comment (digorath) [reply src/screens/DetailScreen.js:78]: as to the more detailed sentence:

the detail screen options are essentially the options with share + bookmark header.

so in the case of the item not being bookmarkable, we need just the options with share.

we should at some point refactor the screenoptions to make them more readable and structured, though.

File: src/components/widgets/ConstructionSiteNewsWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import { DefaultWidget } from './DefaultWidget';
+
+export const ConstructionSiteNewsWidget = ({ text, additionalProps }: WidgetProps) => {
+  const navigation = useNavigation();
+  const refreshTime = useRefreshTime('news-widget', consts.REFRESH_INTERVALS.ONCE_PER_HOUR);
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp, refreshTime });
+
+  const queryVariables = { dataProviderId: additionalProps?.dataProviderId };
```
[the remark this answers is not in the document]
Comment (digorath) [reply src/components/widgets/ConstructionSiteNewsWidget.tsx:22]: correct.

**Review by donni106 — APPROVED** [review]

**Pull request merged**